src/IlluminateMessageRepository/IlluminateUuidV4MessageRepository.php

```diff
@@ -115,4 +121,30 @@ private function yieldMessagesForResult(Collection $result): Generator
 
         return isset($message) ? $message->header(Header::AGGREGATE_ROOT_VERSION) ?: 0 : 0;
     }
+
+    public function paginate(PaginationCursor $cursor): Generator
+    {
+        if ( ! $cursor instanceof OffsetCursor) {
+            throw new LogicException(sprintf('Wrong cursor type used, expected %s, received %s', OffsetCursor::class, get_class($cursor)));
+        }
+
+        $offset = $cursor->offset();
+        $builder = $this->connection->table($this->tableName)
+            ->limit($cursor->limit())
+            ->offset($offset)
+            ->orderBy($this->tableSchema->incrementalIdColumn(), 'ASC');
```

Comment:
What if we add `sortingColumn()` to the tableSchema interface? Would allow for time sorted vs incremental Id vs "custom usecase" sorting. Not sure if added benefit is there, although it fits in the overall tableSchema design.